api.go

```diff
@@ -213,19 +220,58 @@ func (c *Client) LocalInstall(filePath string) (*APIState, error) {
 	}
 	req.FilePath = filePath
 
-	_, body, errs := gorequest.New().Post(buildURL("/local_install")).Send(req).EndStruct(&state)
-	if len(errs) > 0 {
-		return nil, errs[0]
+	response, err := processRequest(string("/local_install"), state, req, "POST")
+	if err != nil {
+		return nil, err
 	}
 
-	err := json.Unmarshal([]byte(body), &state)
+	err = json.Unmarshal([]byte(response), &state)
 	if err != nil {
 		return nil, err
 	}
 
 	return &state, nil
 }
 
+func (c *Client) AbortDownload() (*APIState, error) {
+	var state APIState
+
+	response, err := processRequest(string("/update/download/abort"), state, nil, "POST")
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal([]byte(response), &state)
+	if err != nil {
+		return nil, err
+	}
+
+	return &state, nil
+}
+
+func processRequest(url string, responseStruct Interface, req Interface, method string) ([]byte, error) {
+	switch method {
+	case "GET":
+		_, _, errs := gorequest.New().Get(buildURL(url)).EndStruct(&responseStruct)
+		if len(errs) > 0 {
+			return nil, errs[0]
+		}
+	case "POST":
+		_, _, errs := gorequest.New().Post(buildURL(url)).Send(req).EndStruct(&responseStruct)
+		if len(errs) > 0 {
+			return nil, errs[0]
+		}
+	}
+
+	response, err := json.Marshal(responseStruct)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Why marshal it here and not unmarshal? We could drop Unmarshall on all methods if we move it here, no?

Reply:
The problem was the return type of the function.

Reply:
Well, for this function you can use the `interface {}` no?